applications/search/src/redux/modules/datasets.js

```diff
@@ -0,0 +1,70 @@
+import _ from 'lodash';
+import { normalizeAggregations } from '../../lib/normalizeAggregations';
+import { fetchActions } from '../fetchActions';
+
+export const DATASETS_REQUEST = 'DATASETS_REQUEST';
+export const DATASETS_SUCCESS = 'DATASETS_SUCCESS';
+export const DATASETS_FAILURE = 'DATASETS_FAILURE';
+
+function shouldFetch(metaState, query) {
+  const threshold = 60 * 100; // seconds
+  return (
+    !metaState ||
+    (!metaState.isFetching &&
+      ((metaState.lastFetch || 0) < Date.now() - threshold ||
+        metaState.cachedQuery !== query))
+  );
+}
+
+export function fetchDatasetsIfNeededAction(query) {
+  return (dispatch, getState) =>
+    shouldFetch(_.get(getState(), ['datasets', 'meta']), query) &&
+    dispatch(
+      fetchActions(`/datasets?${query}`, [
+        DATASETS_REQUEST,
+        { type: DATASETS_SUCCESS, meta: { query } },
+        DATASETS_FAILURE
+      ])
+    );
+}
+
+const initialState = {};
+
+export function datasets(state = initialState, action) {
+  switch (action.type) {
+    case DATASETS_REQUEST:
+      return {
+        ...state,
+        meta: {
+          isFetching: true,
+          lastFetch: null,
+          cachedQuery: null
+        }
+      };
+    case DATASETS_SUCCESS:
+      return {
+        ...state,
+        datasetItems: _.get(action.payload, ['hits', 'hits']),
```

Comment:
Så du bestemte å ikke bruke query som lagringsnøkkel 😄 
Da forventer en delete logikk i noen plass, at det er garantert at query og svar står alltid i synk.
(hvis query endres, da må svare slettes). Potensjelle navigasjonproblemer, for eksempel.

hvis du bruker nøklet lagring, da:
- trenger ikke å tenke om slettelogikk
- har du identisk løsning med andre request/response typ redux moduler og kan eventuellt konverteres til lib
- cleanup er veldig enkelt å implementere, trenger ikke bekymre seg om data akkumulasjon

Reply:
Hvis query endres, se på search-reducer. Den dispatcher gjør da alltid i tillegg en history.push som i sin tur trigger en ny fetchDatasets fra search-page. Er ikke det garanti for at query og svar er synket?